index.js

```diff
@@ -274,8 +280,9 @@ exports.hook_data_post = function (next, connection) {
   })
 
   req.on('error', (err) => {
-    if (!connection || !connection.transaction) return;
+    if (!connection.transaction) return nextOnce(); // client gone
```

Comment:
Guard against the possibility that the connection is alive but the transaction has ended.
```suggestion
    if (!connection?.transaction) return nextOnce(); // client gone
```